docs/design/extender.md

```diff
@@ -0,0 +1,64 @@
+# Volcano Framework Extender
+## Summary
+  At present, volcano has not provided a mechanism for non-invasive function realization. If the user needs to implement an internal scheduling function, the source code of volcano must be modified.
+  Such as kube-scheduler, this proposal is to dynamically extend Volcano's scheduling capabilities via an http request.
+## Motivation
+  Currently, the only way to implement a scheduling strategy based on Volcano is to modify Volcano's code and recompile.
+  This method requires developers to deeply understand the working principle of Volcano and the related architecture, which is not friendly to beginners. 
+  By adding other processes and based on network communication, it will bring some performance degradation, but it will also improve scalability. Users can make their own trade-offs between the two.
+### Goals
+  Similar to kube-scheduler, the user-defined scheduling algorithm is registered to volcano through http server.
+### Non-Goals
+  Does not solve the performance degradation caused by the network.
+
+## Design Details
+### Implementation Details
+  Implement a plugin named extender, the plugin will register related functions in the SessionOpen function, and pass the snapshot to the endpoint through the http method.
+  The extender plugin will choose which methods to register according to the configuration file, and is responsible for serializing and deserializing the parameters. When there is an error in the http call, the plugin decides how to deal with it according to the configuration file.
+### Extender Arguments
+```
+- plugins:
+   - name: extender
+     arguments:
+       extender.urlPrefix: http://127.0.0.1
+       extender.httpTimeout: 100ms
+       extender.openSessionOpenVerb: onSessionOpen
```

Comment:
prefer to have default value for those verbs to make configuration simple :)

Reply:
and have a parameter `extender.defaultVerbs = false` to disable all default verbs if necessary.

Reply:
anyway, we can improve that later; as I'm not sure which one is simpler :)

Reply:
If plugin provide default value of verbs, how to distinguish between default value and null value(null value means that the function has not been defined by the user) 
Maybe we can add an example configuration in example package

Reply:
that's ok :)

Reply:
> and have a parameter `extender.defaultVerbs = false` to disable all default verbs if necessary.

that's a good idea. We can do some user interviews to make configuration more simpler